src/Engine/ProtoCore/Lang/Type.cs

```diff
@@ -314,7 +324,13 @@ public string GetType(int UID)
         public int GetType(string ident)
         {
             Validity.Assert(null != classTable);
-            return classTable.IndexOf(ident);
+            var index = classTable.IndexOf(ident);
+            if (index != Constants.kInvalidIndex) return index;
+
+            if(clrToDSTypeMap.TryGetValue(ident, out string dsType))
+                return classTable.IndexOf(dsType);
```

Comment:
while I think this PR looks good - I am curious if you think there could be performance implications here, this `IndexOf` call doesn't seem to be straight forward, it seems to do partial matching? So in the worst case we'll now do 2x the lookups in some pathologically bad cases, do you think will come up rarely, and if so is this method `classTable.IndexOf` linear in complexity or something worse?

Reply:
I think right now, it's only going to take place for a node like `RemoveIfNot `where there is the possibility of checking for a C# type.